HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -61,13 +61,30 @@ HLTScoutingEgammaProducer::HLTScoutingEgammaProducer(const edm::ParameterSet& iC
       HcalPFClusterIsoMap_(consumes<RecoEcalCandMap>(iConfig.getParameter<edm::InputTag>("HcalPFClusterIsoMap"))),
       egammaPtCut(iConfig.getParameter<double>("egammaPtCut")),
       egammaEtaCut(iConfig.getParameter<double>("egammaEtaCut")),
+      absEtaLowEdges(iConfig.getParameter<std::vector<double> >("absEtaLowEdges")),
       egammaHoverECut(iConfig.getParameter<double>("egammaHoverECut")),
+      egammaSigmaIEtaIEtaCut(iConfig.getParameter<std::vector<double> >("egammaSigmaIEtaIEtaCut")),
       mantissaPrecision(iConfig.getParameter<int>("mantissaPrecision")),
       saveRecHitTiming(iConfig.getParameter<bool>("saveRecHitTiming")),
       rechitMatrixSize(iConfig.getParameter<int>("rechitMatrixSize")),  //(2n+1)^2
       rechitZeroSuppression(iConfig.getParameter<bool>("rechitZeroSuppression")),
       ecalRechitEB_(consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("ecalRechitEB"))),
       ecalRechitEE_(consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("ecalRechitEE"))) {
+  // cross-check for compatibility in input vectors
+  if (absEtaLowEdges.size() != egammaSigmaIEtaIEtaCut.size()) {
+    throw cms::Exception("IncompatibleVects")
+        << "absEtaLowEdges and egammaSigmaIEtaIEtaCut vectors should be of the same size. \n";
+  }
+
+  for (unsigned int aIt = 0; aIt < absEtaLowEdges.size() - 1; aIt++) {
```

Comment:
If the size()==0, then the loop condition evaluates to false by default. We do not need a check for the values if the vector is empty, to begin with. If there is a default way to do this, please suggest @missirol. The "-1" in the loop condition is necessary to prevent a seg fault when checking the last entry of the vector.